presto-main/src/main/java/com/facebook/presto/sql/planner/PlanVerifiersFactory.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.StandardErrorCode;
+import com.facebook.presto.sql.analyzer.SemanticErrorCode;
+import com.facebook.presto.sql.analyzer.SemanticException;
+import com.facebook.presto.sql.planner.plan.ApplyNode;
+import com.facebook.presto.sql.planner.plan.FilterNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.ProjectNode;
+import com.facebook.presto.sql.planner.plan.TableScanNode;
+import com.facebook.presto.sql.planner.plan.ValuesNode;
+import com.facebook.presto.sql.tree.DefaultTraversalVisitor;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.SubqueryExpression;
+import com.google.common.collect.ImmutableList;
+
+import javax.inject.Provider;
+
+import java.util.List;
+
+public class PlanVerifiersFactory
+        implements Provider<List<PlanVerifier>>
+{
+    private final List<PlanVerifier> verifiers;
+
+    public PlanVerifiersFactory()
+    {
+        this.verifiers = ImmutableList.of(new NoSubqueryExpressionLeft(),
+                new NoApplyNodeLeft());
+    }
+
+    @Override
+    public List<PlanVerifier> get()
```

Comment:
why `get()` and `verify()` method? One mechanism for verification should be enough, right?